Rendera/include/scene.hpp

```diff
@@ -85,29 +90,34 @@ class Scene{
 
         }
         ~Scene(){
-            delete hitable_sphere;
             delete hitinfo;
         }
-        void set_sphere(int index, vec3 center, float radius, Material mat){
-            if (index < n_sphere)
-                hitable_sphere[index] = sphere(center, radius, mat);
-            else
-                std::cout<<"accessing wrong index"<<std::endl;
+        void add_sphere(vec3 center, float radius, Material mat){
+            n_sphere++;
+            hitable_sphere.push_back(Sphere(center, radius, mat));
+        }
+        void add_cube(vec3 position, vec3 a, vec3 b, vec3 c, Material mat){
+            n_cube++;
+            hitable_cube.push_back(Cube(position, a, b, c, mat));
         }
 
 
 
         void hit_object(int x, int y){
-            hitinfo[y*img_width + x].ray_ = ray(cam.get_ray(x, y));
-            for(int i = 0; i < n_sphere; i++){
+            std::vector<Sphere>::iterator sp;
+            std::vector<Cube>::iterator cb;
```

Comment:
These two lines of code are unnecessary because from C++11 onwards, you can use `auto`.